File: faiss/search_cli.py
import argparse
from text_indexer import TextIndexer

def main():
    parser = argparse.ArgumentParser(description='Text Search CLI')
    parser.add_argument('--update', action='store_true', help='Update the index before searching')
    parser.add_argument('--query', type=str, help='Search query')
    parser.add_argument('--results', type=int, default=3, help='Number of results to return')
    args = parser.parse_args()

    indexer = TextIndexer()

    if args.update:
        print("Updating index...")
        indexer.update_index('data')
        print("Index updated successfully!")

    if args.query:
        print(f"\nSearching for: {args.query}")
        results = indexer.search(args.query, k=args.results)
        
        print("\nResults:")
        for text, score in results:
            print(f"\nScore: {score:.4f}")
            print(f"Text: {text}")

if __name__ == "__main__":
    main()

File: search-cli.js
const TextIndexer = require('./textIndexer');
const yargs = require('yargs/yargs');
const { hideBin } = require('yargs/helpers');

async function main() {
    const argv = yargs(hideBin(process.argv))
        .option('update', {
            alias: 'u',
            type: 'boolean',
            description: 'Update the index before searching'
        })
        .option('query', {
            alias: 'q',
            type: 'string',
            description: 'Search query'
        })
        .option('results', {
            alias: 'r',
            type: 'number',
            default: 3,
            description: 'Number of results to return'
        })
        .help()
        .argv;

    try {
        const indexer = new TextIndexer();
        await indexer.initialize();

        if (argv.update) {
            console.log('Updating index...');
            await indexer.updateIndex('data');
            console.log('Index updated successfully!');
        }

        if (argv.query) {
            console.log(`\nSearching for: ${argv.query}`);
            const results = await indexer.search(argv.query, argv.results);

            console.log('\nResults:');
            for (const { score, text } of results) {
                console.log(`\nScore: ${score.toFixed(4)}`);
                console.log(`Text: ${text}`);
            }
        }
    } catch (error) {
        console.error('Error:', error.message);
        process.exit(1);
    }
}

main().catch(console.error);